### PR TITLE
ci: Update GH API

### LINKED ISSUE #1377: [IT] Add requests to GithubApi.kt
Update `GithubApi.kt` with:
#### GET
* https://docs.github.com/en/free-pro-team@latest/rest/reference/issues#list-repository-issues
* https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-commits
* https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-workflow-runs
#### POST
* https://docs.github.com/en/free-pro-team@latest/rest/reference/issues#create-an-issue
* https://docs.github.com/en/free-pro-team@latest/rest/reference/issues#create-an-issue-comment
#### PATCH
* https://docs.github.com/en/free-pro-team@latest/rest/reference/issues#update-an-issue

### PR DESCRIPTION
**Pull request by pawelpasterz**
Fixes #1377 

This PR adds new requests for flank's GH client. Currently not used. Commit was cherry-picked to make future PR (related to flank scripts/IT) smaller

N/A

### CODE REVIEW

File: flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt [hunk 1]
```diff
@@ -11,49 +12,103 @@ import com.jcabi.github.Releases
...
 
 suspend fun getGitHubIssue(githubToken: String, issueNumber: Int) =
-    Fuel.get("https://api.github.com/repos/Flank/flank/issues/$issueNumber")
+    Fuel.get("$URL_BASE/issues/$issueNumber")
         .appendGitHubHeaders(githubToken)
         .awaitResult(GithubPullRequestDeserializer)
-        .mapClientError { it.toGithubException() }
+        .mapErrorToGithubException()
+
+suspend fun getGitHubIssueList(githubToken: String, parameters: Parameters = emptyList()) =
```
Comment (jan-gogo) [inline flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt:56]: Pls, expose the return type for convenient reading. Also for the rest added functions.

Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt:56]: Good point, thanks for suggestion

File: flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt [hunk 2]
```diff
@@ -11,49 +12,103 @@ import com.jcabi.github.Releases
...
+        .mapErrorToGithubException()
+
+// DELETE
+fun deleteOldTag(tag: String, username: String, password: String) =
+    Fuel.delete(DELETE_ENDPOINT + tag)
+        .authentication()
+        .basic(username, password)
+        .response()
+        .third
+        .mapErrorToGithubException()
```
Comment (piotradamczyk5) [inline flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt:113]: could you add tests?

File: flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt [hunk 3]
```diff
@@ -10,6 +10,8 @@ fun <V : Any, E : FuelError, E2 : Exception> Result<V, E>.mapClientError(transfo
     is Result.Failure -> if (error.response.isClientError) Result.Failure(transform(error)) else this
 }
 
+fun <V : Any, E : FuelError> Result<V, E>.mapErrorToGithubException() = mapClientError { it.toGithubException() }
```
Comment (piotradamczyk5) [inline flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt:13]: `mapErrorToGithubException` -> `mapClientErrorToGithubException`

**Review by piotradamczyk5 — CHANGES_REQUESTED** [review]

File: flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt (hunk 2 above)
Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt:113]: Sure, thanks!

File: flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt (hunk 3 above)
Comment (Sloox) [inline flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt:13]: Are you opposed to actually labeling the V & E to actual names for example:
```suggestion
fun <VALUE : Any, ERROR : FuelError> Result<VALUE , ERROR >.mapErrorToGithubException() = mapClientError { it.toGithubException() }
```
It does nothing but increase readability

File: flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    val body: String = "",
+    val state: IssueState = IssueState.OPEN,
+    val milestone: Int? = null,
+    val labels: List<String> = emptyList(),
+    val assignees: List<String> = emptyList()
+)
+
+@Serializable
+enum class IssueState {
+    @SerialName("open") OPEN,
```
Comment (Sloox) [inline flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt:18]: I want to confirm if the kotlinx serialization is fully compatible with the GSON we are making use of
Else if might result in us changing all the annotations.

**Review by Sloox — CHANGES_REQUESTED** [review]

File: flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt (hunk 4 above)
Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt:18]: I think they are, not 100% sure though. In `flank-scripts` module we do not use GSON.

File: flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt (hunk 3 above)
Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/exceptions/FlankScriptsExceptionMappers.kt:13]: Sure, I'll change it. I kept convention from the other functions. I'll make this change there as well

File: flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt (hunk 4 above)
Comment (Sloox) [reply flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt:18]: I saw here we are DE serializing with GSON so just want to confirm.

Comment (piotradamczyk5) [reply flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt:18]: Wait what? We should not do that

Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/github/objects/GitHubUpdateIssue.kt:18]: @Sloox could you give an example? I am unable to find

**Review by jan-gogo — APPROVED** [review]

**Review by Sloox — APPROVED** [review]
Had a look for the Serialization compatibility. It appears to be fine as far as I am aware.

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**